### ISSUE TITLE
DatePicker: stopPropagation/preventDefualt in bs-days-calendar-view

### ISSUE THREAD

**Issue by Abrissirba** [valor-software/ngx-bootstrap#2772]
I'm using ag-grid and tried to add the datepicker as a filter. It is working but I have a problem that if I use a range filter in ag-grid, which means two input boxes https://www.ag-grid.com/javascript-grid-filtering, ag-grids popup-filter hides when you select a date. This is because ag-grid has an event listener on the document that hides the popup. A solution to this is to add a stopPropagation and preventDefault on the mouseEvent for selectDay() in bs-days-calendar-view. Would it have any undesired side effects to change this?

**Comment by valorkin** [comment]
yes, it will
stopPropagation - will block work of outside click
preventDefault - will not help to you

what you can do is to add one more `(click)` handler to element with datepicker,
and stop event propogation

BTW such things is better to ask at stack overflow ;)

**Comment by Abrissirba** [comment]
Thanks, but I don't think I was able to fully describe my problem. Should have put up a plunker instead... :) 

The problem is not the input element but the bs-datepicker-container and since that is injected dynamically to the body I don't know how I can set a (click) handler to that if you don't mean I should use `document.querSelector('bs-datepicker-container').addEventListener('click', evt => evt.stopPropagation());`  I also don't get why a stopPropagation on an element inside the datepicker should affect clickOutside. I tried to add evt.stopPropagation() to BsDaysCalendarViewComponent.selectDay() and didn't notice any difference in clickOutside.

Sorry about asking here and not on stackoverflow, I find it difficult sometimes to decide where this kind of issues/questions belong
